encord/collection.py

```diff
@@ -0,0 +1,173 @@
+
+    @property
+    def created_at(self) -> Optional[datetime]:
+        """
+        Get the collection creation timestamp
+
+        Returns:
+            Optional[datetime]: The collection creation timestamp.
+        """
+        return self._collection_instance.created_at
+
+    @property
+    def last_edited_at(self) -> Optional[datetime]:
+        """
+        Get the collection last edit timestamp
+
+        Returns:
+            Optional[datetime]: The collection last edit timestamp.
+        """
+        return self._collection_instance.last_edited_at
+
+    @staticmethod
+    def _get_collection(api_client: ApiClient, collection_uuid: UUID) -> "Collection":
+        params = GetCollectionParams(uuids=[collection_uuid])
+        orm_item = api_client.get(
+            "index/collections",
+            params=params,
+            result_type=GetCollectionsResponse,
+        )
+        if len(orm_item.results) > 0:
+            return Collection(api_client, orm_item.results[0])
+        raise AuthorisationError("No collection found")
+
+    @staticmethod
+    def _list_collections(
+        api_client: ApiClient,
+        top_level_folder_uuid: UUID | None,
+        collection_uuid_list: List[UUID] | None,
+        page_size: Optional[int] = None,
+    ) -> "Iterable[Collection]":
+        params = GetCollectionParams(
+            topLevelFolderUuid=top_level_folder_uuid, uuids=collection_uuid_list, pageSize=page_size
+        )
+        paged_items = api_client.get_paged_iterator(
+            "index/collections",
+            params=params,
+            result_type=OrmCollection,
+        )
+        for item in paged_items:
+            yield Collection(api_client, item)
+
+    @staticmethod
+    def _delete_collection(api_client: ApiClient, collection_uuid: UUID) -> None:
+        api_client.delete(
+            f"index/collections/{collection_uuid}",
+            params=None,
+            result_type=None,
+        )
+
+    @staticmethod
+    def _create_collection(
+        api_client: ApiClient, top_level_folder_uuid: UUID, name: str, description: str = ""
+    ) -> UUID:
+        params = CreateCollectionParams(topLevelFolderUuid=top_level_folder_uuid)
+        payload = CreateCollectionPayload(name=name, description=description)
+        orm_item = api_client.post(
+            "index/collections",
+            params=params,
+            payload=payload,
+            result_type=UUID,
+        )
+        return orm_item
+
+    def update_collection(self, name: str | None = None, description: str | None = None) -> None:
+        payload = UpdateCollectionPayload(name=name, description=description)
+        self._client.patch(
+            f"index/collections/{self.uuid}",
+            params=None,
+            payload=payload,
+            result_type=None,
+        )
+
+    def get_items(
+        self,
+        page_size: Optional[int] = None,
+    ) -> Iterable[StorageItem]:
```

Comment:
1. Using name `list_items` seems to be more in line with other methods we have in the sdk.
2. I wonder, if we ever going to have a CollectionItem to store some collection-level info for a given item?
If yes, I'd rather store "list_items" call for these, and call this "list_storage_items" or smth. But if collection item is not a thing, we can keep the name shorter indeed.